lgc/patch/PatchBufferOp.cpp

```diff
@@ -479,14 +479,13 @@ void PatchBufferOp::visitBitCastInst(BitCastInst &bitCastInst) {
 
   m_builder->SetInsertPoint(&bitCastInst);
 
-  Value *const pointer = getPointerOperandAsInst(bitCastInst.getOperand(0));
+  Replacement pointer = getRemappedValue(bitCastInst.getOperand(0));
 
-  Value *const newBitCast =
-      m_builder->CreateBitCast(m_replacementMap[pointer].second, getRemappedType(bitCastInst.getDestTy()));
+  Value *const newBitCast = m_builder->CreateBitCast(pointer.second, getRemappedType(bitCastInst.getDestTy()));
 
-  copyMetadata(newBitCast, pointer);
+  copyMetadata(newBitCast, &bitCastInst);
```

Comment:
There's a deliberate change in behaviour here (and a few other instances in this patch). We copy metadata to the new bitcast from the old bitcast, instead of from the old pointer. This seems more correct to me but doesn't seem to have any effect on the tests I've run.